windows/vtpm_cold_add_remove/cold_add_vtpm.yml

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Handle VM without vTPM device situation"
+  when: not vm_with_vtpm
+  block:
+    - name: "Power off instant cloned VM"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: 'powered-off'
+    - name: "Add key provider on vCenter server"
+      include_tasks: ../../common/vcenter_add_key_provider.yml
+      vars:
+        vc_cert_path: "{{ current_test_log_folder }}"
+      when:
+        - key_provider_type is defined and key_provider_type
+        - not (key_provider_added is defined and key_provider_added)
+    - name: "Add vTPM device to cloned VM"
+      include_tasks: ../../common/vm_add_remove_vtpm.yml
+      vars:
+        vtpm_operation: 'present'
+    - name: "Pause 10 seconds after adding vTPM device"
+      ansible.builtin.pause:
+        seconds: 10
+
+- name: "Power on cloned VM"
+  include_tasks: ../../common/vm_set_power_state.yml
+  vars:
+    vm_power_state_set: 'powered-on'
+
+- name: "Get cloned VM primary MAC address"
+  include_tasks: ../../common/vm_wait_primary_nic_mac.yml
+
+- name: "Print VM primary MAC address"
+  ansible.builtin.debug:
+    msg:
+      - "Cloned VM primary MAC address: {{ vm_primary_nic_mac }}"
+      - "Parent VM primary MAC address: {{ vm_nic_mac_clone_from }}"
+
+- name: "Initialize cloned VM IP address"
+  ansible.builtin.set_fact:
+    retry_vm_guest_ip: ""
+- name: "Retry to get cloned VM IP address"
+  block:
+    - name: "Get cloned VM IP address"
+      include_tasks: retry_vm_get_ip.yml
+      with_list: "{{ range(1, 11) | list }}"
+      loop_control:
+        loop_var: retry_count
+  when: >
+    (not retry_vm_guest_ip) or
+    (retry_vm_guest_ip == vm_ip_clone_from)
+
+- name: "Print VM IP address"
```

Comment:
It's better to add a Alert task to check if new IP is got correctly, i.e. different from parent IP.